Fastly/fastly/connector_fastly_waf_audit.py

```diff
@@ -0,0 +1,32 @@
+import datetime
+
+from dateutil.parser import isoparse
+from pydantic import Field
+
+from .connector_fastly_base import FastlyBasicConnectorConfiguration, FastlyConnectorBase
+
+
+class FastlyWAFAuditConnectorConfiguration(FastlyBasicConnectorConfiguration):
+    site: str | None = Field(None, description="Site name", pattern=r"^[0-9a-z_.-]+$")
+
+
+class FastlyWAFAuditConnector(FastlyConnectorBase):
+    configuration: FastlyWAFAuditConnectorConfiguration
+
+    def get_datetime_from_item(self, item: dict) -> datetime.datetime:
+        return isoparse(item["created"])
+
+    def get_next_url(self, from_datetime: datetime.datetime) -> str:
+        from_timestamp = int(from_datetime.timestamp())
+
+        if self.configuration.site:
+            return (
+                f"{self.base_uri}/api/v0/corps/{self.configuration.corp}/sites/{self.configuration.site}/activity?"
+                f"sort=asc&limit={self.configuration.chunk_size}&from={from_timestamp}"
+            )
+
+        else:
+            return (
+                f"{self.base_uri}/api/v0/corps/{self.configuration.corp}/activity?"
+                f"sort=asc&limit={self.configuration.chunk_size}&from={from_timestamp}"
+            )
```

Comment:
Corpos and sites [have different types of events](https://docs.fastly.com/en/ngwaf/monitoring-account-activity-with-audit-logs#corp-audit-log), I'm not sure it make sense to mix them altogether. But, perhaps, it would be useful to get events from multiple sites in one connector

Reply:
We already had different kind of events retrieved from the same connector (ex: Darktrace).